env_spec.py

```diff
@@ -1,53 +1,108 @@
 import re
 
-
-def check_input(field_array):
+valid_types_list = [
+    "color",
+    "date",
+    "datetime-local",
+    "email",
+    "month",
+    "number",
+    "password",
+    "tel",
+    "text",
+    "time",
+    "url",
+    "week",
+]
+
+
+def check_input(input_field):
     """
-    Checks syntax of input(uppercase latin chars,numeric digits, underscores).
+    Checks syntax of input field(uppercase latin chars,numeric digits, underscores).
     Returns False for error or True for correct syntax.
     """
     alphanumeric_that_does_not_start_with_digit = r"^[A-Z_][0-9A-Z_]*$"
 
-    for line in field_array:
-        ret = re.match(alphanumeric_that_does_not_start_with_digit, line)
+    ret = re.match(alphanumeric_that_does_not_start_with_digit, input_field)
 
-        if ret is None:
-            # print("SYNTAX ERROR:WRONG SYMBOLS!")
-            return False
+    if ret is None:
+        return False
 
     return True
 
 
-def render_env_var_spec(env_var_name):
-    env_var_name_lower = env_var_name.lower()
-
-    ret_str = f'<label for="env_spec_{env_var_name_lower}"> {env_var_name}</label>'
-    ret_str += "\n"
-    ret_str += (
-        f'<input id="env_spec_{env_var_name_lower}" name="{env_var_name_lower}" />'
-    )
+def check_type(input_type):
+    """
+    Checks if given type is valid and returns True, else returs False
+    """
+    if input_type not in valid_types_list:
+        return False
+    return True
 
-    return ret_str
 
+def render_env_var_spec(env_var_field, env_var_type):
+    """
+    Creates html string from env_variables (field, type). Returns the string.
+    """
+    env_var_field_lower = env_var_field.lower()
+
+    if env_var_type != "":
+        ret_str = (
+            f'<label for="env_spec_{env_var_field_lower}"> {env_var_field}</label>'
+        )
+        ret_str += "\n"
+        ret_str += f'<input id="env_spec_{env_var_field_lower}" name="{env_var_field_lower}" />'
+        ret_str += "\n"
+        ret_str += f'<input id="env_spec_{env_var_field_lower}" name="{env_var_field_lower}" type="{env_var_type}" />'
+    else:
+        ret_str = (
+            f'<label for="env_spec_{env_var_field_lower}"> {env_var_field}</label>'
+        )
 
-def render_env_spec_to_html(field_array):
-    if check_input(field_array):
-        html_output = []
+    return ret_str
 
-        for field in field_array:
-            html_output.append(render_env_var_spec(field))  # create html output
 
-        return "\n".join(html_output)
-    return ""
+def render_env_spec_to_html(input_str):
+    """
+    Takes the whole string and splits it first by \n and then by :, if possible. Returns html output or "".
+    """
+    html_output = ""
+    last_string_flag = False
+
+    while 1:
+        try:
+            str, rest_str = input_str.split("\n", 1)
```

Comment:
Please rename `str` to something else, since [`str` is a built-in Python function](https://docs.python.org/3/library/functions.html#func-str).

Could `line` make the cut?